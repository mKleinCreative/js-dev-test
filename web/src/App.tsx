import React from 'react';
import AllRepos from './components/AllRepos';
import { Row } from 'react-materialize';

function App() {
  return (
    <div>
      <header>
        <Row>
          <AllRepos />
        </Row>
      </header>
    </div>
  );
}

export default App;
