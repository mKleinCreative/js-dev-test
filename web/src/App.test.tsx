import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';



it('renders correctly', () => {
  const tree = renderer
    .create(<App />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders Navbar', () => {
  render(<App />);
  const linkElement = screen.getByText(/Filter by language/i);
  expect(linkElement).toBeInTheDocument();
});

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<App />, div)
  ReactDOM.unmountComponentAtNode(div)
})