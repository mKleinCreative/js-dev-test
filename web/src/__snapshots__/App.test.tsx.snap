// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header>
    <div
      className="row"
    >
      <div
        className="col s12 m10 offset-m1"
      >
        <nav
          className=""
        >
          <div
            className="nav-wrapper"
          >
            <div
              className="brand-logo"
            >
              Filter by language
            </div>
            <a
              className="sidenav-trigger"
              data-target="mobile-nav"
              href="#!"
            >
              <i
                className="material-icons"
              >
                menu
              </i>
            </a>
            <ul
              className="hide-on-med-and-down right"
            >
              <li>
                <a
                  onClick={[Function]}
                >
                  None
                </a>
              </li>
            </ul>
          </div>
        </nav>
        <ul
          className="sidenav right"
          id="mobile-nav"
        >
          <li>
            <a
              onClick={[Function]}
            >
              None
            </a>
          </li>
        </ul>
        <div>
          Loading...
        </div>
      </div>
    </div>
  </header>
</div>
`;
